### DISCUSSION TITLE
Issue with gmail Confirm Your Signup

### DISCUSSION THREAD

**Discussion by tbenci** [supabase/supabase#18380] [Questions]
Hi all,

I am using the client side supabase.auth.signUp.

```
const onSubmit = handleSubmit(async (data) => {
    try {
      const {
        data: { user },
        error,
      } = await supabase.auth.signUp({
        email: data.email,
        password: data.password,
        options: {
          emailRedirectTo: `${location.origin}${routeNames.authCallBack}`,
        },
      });
      if (error) throw error;
      if (user) {
        if (user.identities && user.identities.length === 0) {
          toast.error("This user already exists");
        } else {
          toast.success("Check your inbox for a confirmation email");
          router.push(routeNames.signin);
        }
      } else {
        throw "Unknown error experienced";
      }
    } catch (error) {
      toast.error(handleError(error));
    }
  });
```

My call back is out of the box from Supabase

```
// System dependencies

import type { NextRequest } from "next/server";
import { NextResponse } from "next/server";
import { cookies } from "next/headers";
import { createRouteHandlerClient } from "@supabase/auth-helpers-nextjs";

// Application dependencies

import type { Database } from "@/lib/database.types";

export async function GET(request: NextRequest) {
  const requestUrl = new URL(request.url);
  const code = requestUrl.searchParams.get("code");

  if (code) {
    const cookieStore = cookies();
    const supabase = createRouteHandlerClient<Database>({ cookies: () => cookieStore });
    await supabase.auth.exchangeCodeForSession(code);
  }

  return NextResponse.redirect(requestUrl.origin);
}
```
When a gmail user receives their confirm email and clicks the link strange things happen.

1. The user account on supabase is waiting for verification

![image](https://github.com/supabase/supabase/assets/7336557/639abaf9-2de1-48bb-9ee0-d3a56905d383)

2. When they click the link the account is verified 

![image](https://github.com/supabase/supabase/assets/7336557/060b9a3c-7343-4074-9eba-9ecf054550a4)

4. But the redirect receives an error message.

`http://localhost:3000/signin#error=unauthorized_client&error_code=401&error_description=Email+link+is+invalid+or+has+expired`

5. The thing is that the user sees an error because I am decoding the hash parameter's error_description into a toast BUT the user account is verified.

It seems that I get 95% of the way there with supabase but then something like this happens and I want to throw it all away. Any help or ideas appreciated.

Cheers
Tony

**Comment by GaryAustin1** [comment]
Probably this:  https://supabase.com/docs/guides/auth/auth-email-templates#limitations

**Reply by tbenci** [reply]
That probably explains why OPT has fallen in a hole on gmail as well. I need to find a good reference on how to do that I suppose. Anyone know of one?

You'd reckon that if some feature was not going to work with gmail there might be a prominent solution to the issue.
